Flaky test with potential data race in autoschema

Test is flaky, so it does not fail every time:

I have not yet investigated if this is only an issue in the test setup or if there is actually a potential data race happening in the autoschema feature.

```
==================
WARNING: DATA RACE
Write at 0x00c00022e880 by goroutine 69:
  github.com/semi-technologies/weaviate/usecases/objects.(*fakeSchemaManager).AddClass()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/fakes_for_test.go:76 +0x274
  github.com/semi-technologies/weaviate/usecases/objects.(*autoSchemaManager).createClass()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/auto_schema.go:97 +0x33c
  github.com/semi-technologies/weaviate/usecases/objects.(*autoSchemaManager).performAutoSchema()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/auto_schema.go:68 +0x33c
  github.com/semi-technologies/weaviate/usecases/objects.(*autoSchemaManager).autoSchema()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/auto_schema.go:51 +0xba
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).validateObject()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:101 +0x132
Previous read at 0x00c00022e880 by goroutine 70:
  github.com/semi-technologies/weaviate/entities/schema.GetClassByName()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/entities/schema/backward_compat.go:49 +0x51
  github.com/semi-technologies/weaviate/entities/schema.(*Schema).FindClassByName()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/entities/schema/accessors.go:29 +0xfa
  github.com/semi-technologies/weaviate/usecases/objects.(*vectorObtainer).getVectorizerOfClass()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/obtain_vector.go:90 +0xbb
  github.com/semi-technologies/weaviate/usecases/objects.(*vectorObtainer).Do()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/obtain_vector.go:44 +0xb3
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).validateObject()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:140 +0x71b
Goroutine 69 (running) created at:
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).validateObjectsConcurrently()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:84 +0x210
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).addObjects()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:53 +0x293
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).AddObjects()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:44 +0x378
  github.com/semi-technologies/weaviate/usecases/objects.Test_BatchManager_AddObjects_WithNoVectorizerModule.func5()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add_test.go:136 +0x5a4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1193 +0x202
Goroutine 70 (finished) created at:
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).validateObjectsConcurrently()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:84 +0x210
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).addObjects()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:53 +0x293
  github.com/semi-technologies/weaviate/usecases/objects.(*BatchManager).AddObjects()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add.go:44 +0x378
  github.com/semi-technologies/weaviate/usecases/objects.Test_BatchManager_AddObjects_WithNoVectorizerModule.func5()
      /Users/etiennedilocker/go/src/github.com/semi-technologies/weaviate/usecases/objects/batch_add_test.go:136 +0x5a4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1193 +0x202
==================
--- FAIL: Test_BatchManager_AddObjects_WithNoVectorizerModule (0.00s)
    --- FAIL: Test_BatchManager_AddObjects_WithNoVectorizerModule/with_objects_without_IDs_and_nonexistent_class_and_auto_schema_enabled (0.00s)
        testing.go:1092: race detected during execution of test
    testing.go:1092: race detected during execution of test
FAIL
coverage: 63.0% of statements
```

--- a/usecases/objects/auto_schema.go
+++ b/usecases/objects/auto_schema.go
@@ -55,6 +55,8 @@ func (m *autoSchemaManager) autoSchema(ctx context.Context, principal *models.Pr
 
 func (m *autoSchemaManager) performAutoSchema(ctx context.Context, principal *models.Principal,
 	object *models.Object) error {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
 	if len(object.Class) == 0 {
 		// stop performing auto schema
 		return fmt.Errorf(validation.ErrorMissingClass)
@@ -72,8 +74,6 @@ func (m *autoSchemaManager) performAutoSchema(ctx context.Context, principal *mo
 
 func (m *autoSchemaManager) getClass(principal *models.Principal,
 	object *models.Object) (*models.Class, error) {
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
 	s, err := m.schemaManager.GetSchema(principal)
 	if err != nil {
 		return nil, err
@@ -84,8 +84,6 @@ func (m *autoSchemaManager) getClass(principal *models.Principal,
 
 func (m *autoSchemaManager) createClass(ctx context.Context, principal *models.Principal,
 	className string, properties []*models.Property) error {
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
 	class := &models.Class{
 		Class:       className,
 		Properties:  properties,
@@ -99,8 +97,6 @@ func (m *autoSchemaManager) createClass(ctx context.Context, principal *models.P
 
 func (m *autoSchemaManager) updateClass(ctx context.Context, principal *models.Principal,
 	className string, properties []*models.Property, existingProperties []*models.Property) error {
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
 	propertiesToAdd := []*models.Property{}
 	for _, prop := range properties {
 		found := false

